app/models/apple/episode.rb

```diff
@@ -114,26 +143,38 @@ def self.publish(api, episodes)
         episodes.map(&:publish_episode_bridge_params))
     end
 
-    def self.insert_sync_logs(episodes, results)
+    def self.upsert_sync_logs(episodes, results)
       episodes_by_guid = episodes.map { |ep| [ep.guid, ep] }.to_h
 
       results.map do |res|
-        apple_id = res.dig("api_response", "val", "data", "id")
-        guid = res.dig("api_response", "val", "data", "attributes", "guid")
-        ep = episodes_by_guid.fetch(guid)
+        upsert_sync_log(episodes_by_guid[res.dig("request_metadata", "guid")], res)
+      end
+    end
+
+    def self.upsert_sync_log(ep, res)
```

Comment:
This mirrors `Apple::PodcastContainer.upsert_podcast_container`, `Apple::PodcastDelivery.upsert_podcast_delivery`, etc.

But since creating the episode is handled a bit differently, this falls back on a single responsibility of upserting the apple sync log (remote data wrapper).